src/main/kotlin/com/dp/advancedgunnerycontrol/weaponais/tags/ConservePDAmmoTag.kt

```diff
@@ -0,0 +1,29 @@
+package com.dp.advancedgunnerycontrol.weaponais.tags
+
+import com.dp.advancedgunnerycontrol.weaponais.*
+import com.fs.starfarer.api.combat.CombatEntityAPI
+import com.fs.starfarer.api.combat.MissileAPI
+import com.fs.starfarer.api.combat.ShipAPI
+import com.fs.starfarer.api.combat.WeaponAPI
+import org.lwjgl.util.vector.Vector2f
+
+// Only fire at full ROF if target is missile or fighter and ammo < ammoThreshold
+class ConservePDAmmoTag(weapon: WeaponAPI, private val ammoThreshold: Float) : WeaponAITagBase(weapon) {
+
+    override fun isBaseAiValid(entity: CombatEntityAPI): Boolean = true
```

Comment:
So this tag is actually working as expected, it just might not be clear what it does.

The problem I noticed is that if I have PD weapons that use ammo, I can either set them to PD, and then they have no utility outside of PD, or I don't set them to PD and then they blow all their ammo on ships and are less effective when a bunch of fighters comes.

This tag will restrict targetting to fighters and missiles when ammo is below 90%, thus allowing for some utility against ships whilst still maintaining reserves for fighter swarms. I have tested it and it does work as intended. I usually couple it with prioritisePD tag instead of enforcing any targeting priorities.